File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        res = []
        for i in range(len(nums)):
            if target - nums[i] in nums[i+1:]:
                res.append(nums.index(nums[i]))
                res.append(i +1+ nums[i+1:].index(target - nums[i]))
        return res

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> umap; 
        for (int i=0;i<nums.size();i++)
        {
            int secondNumber = target - nums[i];
            if(umap.find(secondNumber)!=umap.end() && umap[secondNumber] != i)
            {
                return {i,umap[secondNumber]};
            }
            umap[nums[i]] = i;
        }
        return {};
        
    }
};